framework/src/media/audio/audio_manager.c

```diff
@@ -159,6 +161,31 @@ static audio_manager_result_t set_audio_volume(audio_card_type_t type, uint8_t v
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+static audio_manager_result_t init_audio_stream(audio_card_type_t card_type)
+{
+	int i;
+	int max_card_num = CONFIG_AUDIO_MAX_INPUT_CARD_NUM;
+	audio_card_info_t *card_info = g_audio_in_cards;
+
+	if (card_type == OUTPUT) {
+		max_card_num = CONFIG_AUDIO_MAX_OUTPUT_CARD_NUM;
+		card_info = g_audio_out_cards;
+	}
+
+	for (i=0; i < max_card_num; i++) {
```

Comment:
We don't have to init none-existing card.
'if_found_then_init' is correct, I think

Reply:
Modified.